python/vyos/firewall.py

```diff
@@ -16,9 +16,71 @@
 
 import re
 
+from vyos.util import call
 from vyos.util import cmd
 from vyos.util import dict_search_args
 
+
+# Functions for firewall group domain-groups
+def get_ips_domains_dict(list_domains):
+    """
+    Get list of IPv4 addresses by list of domains
+    Ex: get_ips_domains_dict(['ex1.com', 'ex2.com'])
+        ['192.0.2.1', '192.0.2.2', '192.0.2.3']
+    """
+    from socket import gethostbyname_ex
+    from socket import gaierror
+
+    ip_list = []
+    for domain in list_domains:
+        try:
+            _, _, ips = gethostbyname_ex(domain)
+            for entry in ips:
+                ip_list.append(entry)
+        except gaierror:
+            pass
```

Comment:
I wonder if instead of silently continuing, the update should be halted and values remain in the current state? Prevents any sets being blanked when resolution fails, with the last known good IPs staying put.

Reply:
If it is first boot? And dhcp or pppoe not yet assigned

Reply:
I like the idea and the explanation from @sarthurdev seems reasonable.

Reply:
What should I change in this function?
It is not clear to me.

Reply:
PR'd your fork: https://github.com/sever-sever/vyos-1x/pull/2

Reply:
> PR'd your fork: [sever-sever#2](https://github.com/sever-sever/vyos-1x/pull/2)

Thanks, done!